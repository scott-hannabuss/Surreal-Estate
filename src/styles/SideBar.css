.sidebar {
  height: 80%;
  width: 15%;
  position: fixed;
  background-color: white;
  left: 0;
  padding-top: 20px;
  font-size: 10px;
  text-align: left;
  background-color: white;
}

.sidebar a {
  font-family: "Lato", sans-serif;
  text-decoration: none;
  text-decoration: underline;
  font-size: 150%;
  display: block;
  margin-bottom: 20%;
  padding-right: 200%;
}

.input_sidebar {
  width: 80%;
  margin-bottom: 7%;
  display: block;
  padding: 0.5%;
  background-color: #12093b;
  border-radius: 25px;
  font-family: "Lato", sans-serif;
  text-align: center;
  margin-top: 5%;
}

.sidebar_search-button {
  color: black;
  justify-content: center;
  align-content: center;
  display: grid;
  padding: 0.5%;
  width: 50%;
  border-radius: 25px;
  font-family: "Lato", sans-serif;
}

.sidebar_header {
  color: black;
  margin-top: 25%;
  margin-bottom: 20%;
  font-family: "Lato", sans-serif;
  font-size: 150%;
}

@media only screen and (min-width: 751px) and (max-width: 1025px) {
  .sidebar {
    text-align: center;
    justify-content: space-between;
    width: 25%;
    margin-top: 1px;
    font-size: 0.5em;
    background-color: white;
    position: fixed;
    text-decoration: none;
  }
  .sidebar_header {
    margin-right: 35%;
    margin-bottom: 20%;
  }
  .sidebar a {
    margin-right: 10%;
    margin-bottom: 20%;
    text-decoration: none;
    text-decoration: underline;
    color: black;
  }
}
