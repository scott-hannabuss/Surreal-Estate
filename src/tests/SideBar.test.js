import React from "react";
import { render } from "@testing-library/react";
import SideBar from "../components/Properties";
import { BrowserRouter } from "react-router-dom";

describe("Sidebar", () => {
  it("renders sidebar as expected", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <SideBar />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

it("renders a filter by city header", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <SideBar />
    </BrowserRouter>
  );
  const cityHeader = getByTestId("test-filter-by-city-header");
  expect(cityHeader).toBeTruthy();
});

it("renders a filter by price header", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <SideBar />
    </BrowserRouter>
  );
  const priceHeader = getByTestId("test-filter-by-price-header");
  expect(priceHeader).toBeTruthy();
});
