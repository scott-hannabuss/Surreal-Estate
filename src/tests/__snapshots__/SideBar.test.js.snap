// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders sidebar as expected 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sidebar"
    />
    <div
      class="sidebar"
    >
      <ul
        class="sidebar_links"
      >
        <h3
          class="sidebar_header"
          datatest-id="test-filter-by-city-header"
        >
          Filter By City
        </h3>
        <a
          href="/?query={\\"city\\":\\"Manchester\\"}"
        >
          Manchester
        </a>
        <a
          href="/?query={\\"city\\":\\"Leeds\\"}"
        >
          Leeds
        </a>
        <a
          href="/?query={\\"city\\":\\"Sheffield\\"}"
        >
          Sheffield
        </a>
        <a
          href="/?query={\\"city\\":\\"Liverpool\\"}"
        >
          Liverpool
        </a>
        <h3
          class="sidebar_header"
          datatest-id="test-filter-by-price-header"
        >
          Filter by Price
        </h3>
        <a
          href="/?sort={\\"price\\":-1}"
        >
          Price Descending
        </a>
        <a
          href="/?sort={\\"price\\":1}"
        >
          Price Ascending
        </a>
      </ul>
    </div>
    <div />
    <div
      class="properties-page"
    >
      <h1
        class="header_properties-page"
      >
        View Current Properties
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
